middleware/auth.go

```diff
@@ -187,3 +187,22 @@ func (a *Authenticator) basicAdminUser(r *http.Request) bool {
 
 	return true
 }
+
+// RBAC middleware allows role based control for routes
+// this handler internally wrapped with auth(true) to avoid situation if RBAC defined without prior Auth
+func (a *Authenticator) RBAC(role string, next http.Handler) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		user, err := token.GetUserInfo(r)
+		if err != nil {
+			http.Error(w, "Unauthorized", http.StatusUnauthorized)
+			return
+		}
+
+		if role != user.Role {
```

Comment:
maybe EqualFold will be a better one here? Not 100% sure, but case sensetivty in roles probably not needed